### PR TITLE
Fix max_formspec_size not taking gui_scaling into account

### PR DESCRIPTION
**Pull request by grorp**
The `max_formspec_size` value returned by `minetest.get_player_window_information()` is currently incorrect if the player has set the `gui_scaling` setting to a value other than 1.

That's because `prefer_imgsize` in `GUIFormSpecMenu::regenerateGui`  (<a href="https://github.com/minetest/minetest/blob/65692ad1b556c4e7b7e815e24270c869e8c20adf/src/gui/guiFormSpecMenu.cpp#L3250-L3257">link to the code</a>) is scaled by `gui_scaling`, but `ClientDynamicInfo::calculateMaxFSSize` (<a href="https://github.com/minetest/minetest/blob/65692ad1b556c4e7b7e815e24270c869e8c20adf/src/clientdynamicinfo.h#L38-L51">link to the code</a>) doesn't care. This PR just adds a division by `gui_scaling` to `calculateMaxFSSize`.

This bug cannot be worked around by mods since mods don't get the `gui_scaling`, but only the `real_gui_scaling` (`gui_scaling` * display density) of the player. Therefore, fixing this bug cannot not break any mods that already work around it.

## To do

This PR is a Ready for Review.

## How to test

Set `gui_scaling` to 0.5.
On the master branch, start Devtest and do `/testfullscreenfs`. The formspec won't be shown fullscreen.
Do the same on this PR. The formspec will be shown fullscreen.

### CODE REVIEW

File: src/client/game.cpp [hunk 1]
```diff
@@ -2594,12 +2594,13 @@ ClientDynamicInfo Game::getCurrentDynamicInfo() const
 {
 	v2u32 screen_size = RenderingEngine::getWindowSize();
 	f32 density = RenderingEngine::getDisplayDensity();
-	f32 gui_scaling = g_settings->getFloat("gui_scaling") * density;
-	f32 hud_scaling = g_settings->getFloat("hud_scaling") * density;
+	f32 gui_scaling = g_settings->getFloat("gui_scaling");
```
Comment (SmallJoker) [inline src/client/game.cpp:2597]: Needs range limiting ( > 0) to not cause issues in `calculateMaxFSSize`. Furthermore: wouldn't it make sense to move these values to variables in `ClientDynamicInfo`? Currently there's a code duplication that has potential to go out of sync easily due to different code areas: game client and main menu.

File: src/clientdynamicinfo.h [hunk 2]
```diff
@@ -20,21 +20,39 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
 
 struct ClientDynamicInfo
 {
+public:
 	v2u32 render_target_size;
 	f32 real_gui_scaling;
 	f32 real_hud_scaling;
 	v2f32 max_fs_size;
 
+	static const ClientDynamicInfo get_current() {
```
Comment (rubenwardy) [inline src/clientdynamicinfo.h:35]: This class is used on the server too. It shouldn't know about the rendering engine

**Review by SmallJoker — APPROVED** [review]
The GUI looks as usual, thus LGTM.

File: src/clientdynamicinfo.h [hunk 3]
```diff
@@ -35,17 +40,36 @@ struct ClientDynamicInfo
 				abs(real_hud_scaling - other.real_hud_scaling) < 0.001f;
 	}
 
-	static v2f32 calculateMaxFSSize(v2u32 render_target_size) {
+#ifndef SERVER
+	static const ClientDynamicInfo getCurrent() {
```
Comment (Desour) [inline src/clientdynamicinfo.h:44]: Why is the return type const?

**Review by Desour — COMMENTED** [review]
Tested, works.

One thing I noticed: When I'm in window mode, the fullscreenfs is still surrounded by a small boarder. (But that's prolly not relevant for this PR.)

File: src/clientdynamicinfo.h (hunk 3 above)
Comment (grorp) [reply src/clientdynamicinfo.h]: I don't know. I've now undone that change.

**Comment by rubenwardy** [comment]
Hi, please wait for me to test this

**Review by rubenwardy — APPROVED** [review]
works well

**Comment by grorp** [comment]
Bump. This has now also been approved by rubenwardy, so I think it should be ready for merging.

**Pull request merged**